service.go

```diff
@@ -282,7 +288,19 @@ func NewService(serviceName string, serviceVersion string, confProfile string, c
 	svc = &Service{}
 	svc.startTime = startTime
 	svc.svcInfo = &config.Service
-	common.Driver = proto
+	if casted, ok := proto.(dsModels.ProtocolDriver); ok {
+		common.Driver = casted
+	} else {
+		err := fmt.Errorf("ProtocolDriver must be implemented\n")
+		return nil, err
+	}
+
+	if casted, ok := proto.(dsModels.ProtocolDiscovery); ok {
+		common.Discovery = casted
+	} else {
+		common.Discovery = nil
+		common.LoggingClient.Warn("ProtocolDiscovery not implemented")
```

Comment:
Use INFO instead of WARN to log the message.
Also, add the verb into the sentence

> ProtocolDiscovery **is** not implemented